BredaUniversityGames/Y2024-25-PR-BB
# Feature/steam api

### Description

- Added Steam module which adds basic Steam functionality (for example Steam Overlay)
- Added Steamworks API files with the needed DLL

https://github.com/user-attachments/assets/4be18e04-a039-4a84-a305-ccd70721155d

`steam_appid.txt` file will contain our app ID steam gives us. This is needed when running the application outside of Steam (for example CLion). This file is important, as it must not be distributed when uploading the build to Steam.

### Issues

[Codecks card](https://bubonic-brotherhood.codecks.io/milestones/run/40/card/18o-steam-api)

### Test criteria

When Steam is open:

1. Make sure Steam is open
2. Launch the program via CLion as normally
3. When the game is launched, notice the steam popup in the right bottom corner
4. Press Shift + Tab to open the Steam Overlay

When Steam is closed:

1. Launch the program
2. A warning pops up in the console -> [Steamworks] Steam is probably not running, Steamworks API functionality will be unavailable
3. The game can be played as normal, but Seam features are not available
